lib/FilesHooks.php

```diff
@@ -463,6 +478,12 @@ protected function fileMoving($oldPath, $newPath) {
 		$affectedUsers = $accessList['users'];
 		$oldUsers = $this->oldAccessList['users'];
 
+		// file can be shared using GroupFolders, including ACL check
+		if ($this->config->getSystemValueBool('activity_use_cached_mountpoints', false)) {
+			$this->userMountCache->clear(); // clear cache for new data
+			$affectedUsers = array_merge($affectedUsers, $this->getAffectedUsersFromCachedMounts($fileId));
```

Comment:
Why merge? Isn't `getAffectedUsersFromCachedMounts` supposed to return an updated list?

Reply:
Same question for other changes.

Reply:
the list is cached by UserMountCache. because we are moving files, we only get old data from pre-move; clearing the cache allows to get fresh data post-move